waspc/cli/Command/Start.hs

```diff
@@ -2,69 +2,51 @@ module Command.Start
+import           StrongPath               ((</>))
 
 
+-- | Does initial compile of wasp code and then runs the generated project.
+-- It also listens for any file changes and recompiles and restarts generated project accordingly.
 start :: Command ()
 start = do
     waspRoot <- findWaspProjectRootFromCwd
-    waspFile <- findWaspFile waspRoot
     let outDir = waspRoot </> Common.dotWaspDirInWaspProjectDir </> Common.generatedCodeDirInDotWaspDir
-    let options = CompileOptions
-            { externalCodeDirPath = waspRoot </> Common.extCodeDirInWaspProjectDir }
 
-    -- TODO: This just compiles once. We need `wasp start` to do much more.
-    waspSays "Compiling wasp code..."
-    errorOrResult <- liftIO $ Lib.compile waspFile outDir options
-    case errorOrResult of
+    waspSaysC "Compiling wasp code..."
+    compilationResult <- liftIO $ compile waspRoot outDir
+    case compilationResult of
         Left compileError -> throwError $ CommandError $ "Compilation failed: " ++ compileError
-        Right () -> waspSays "Code has been successfully compiled.\n"
+        Right () -> waspSaysC "Code has been successfully compiled, project has been generated.\n"
 
-    -- TODO: Do smart install -> if we need to install stuff, install it.
+    -- TODO: Do smart install -> if we need to install stuff, install it, otherwise don't.
     --   This should be responsibility of Generator, it should tell us how to install stuff.
     --   But who checks out if stuff needs to be installed at all? That should probably be
     --   Generator again. After installation, it should return some kind of data that describes that installation.
     --   Then, next time, we give it data we have about last installation, and it uses that
     --   to decide if installation needs to happen or not. If it happens, it returnes new data again.
     --   Right now we have setup/installation being called, but it has not support for being "smart" yet.
-    waspSays "Setting up generated project..."
+    waspSaysC "Setting up generated project..."
     setupResult <- liftIO $ Lib.setup outDir
     case setupResult of
         Left setupError -> throwError $ CommandError $ "Setup failed: " ++ setupError
-        Right () -> waspSays "Setup successful.\n"
-
-    -- TODO: Check node version and then run `npm start` on both web and server.
-    --   Again, this is something that Generator should be responsible for, since it knows how the code is generated.
-    --   It should tell us how to start stuff and we just start it. It should even do composing of the outputs,
-    --   since it knows more than us about that.
-    waspSays "Starting up generated project..."
-    startResult <- liftIO $ Lib.start outDir
-    case startResult of
-        Left startError -> throwError $ CommandError $ "Start failed: " ++ startError
-        Right () -> error "This should never happen, start should never end."
-
-    -- TODO: Listen for changes, if they happen, re-generate the code.
-  where
```

Comment:
aha here it was. It would be cool if Github was smart enough to say "this function was just moved."